rtaudio doesn't build on windows using CMake and VS 2017

I am using CMake and Visual Studio 2017 on Windows 10.
The following linker errors are shown when building target rtaudio:
>exports.def : error LNK2001: unresolved external symbol =
>D:/rtaudio/build/Debug/rtaudiod.lib : fatal error LNK1120: 1 unresolved externals
>Done building project "rtaudio.vcxproj" -- FAILED.

This is the commit that broke it: https://github.com/thestk/rtaudio/commit/9802c1285962f4bb187ee280a21908dc134083d0
The commit before that commit worked.

There isn't any other information shown. I don't know why the linker wants to find a symbol named `=`.
This is how I call cmake:
```bash
mkdir build
cd build
cmake -D CMAKE_INSTALL_PREFIX=./install ..
cmake --build . --target install
```

got the same VC2017 32 bits
```
[100%] Linking CXX shared library rtaudio.dll
LINK Pass 1: command "C:\PROGRA~2\MIB055~1\2017\COMMUN~1\VC\Tools\MSVC\1415~1.267\bin\Hostx86\x86\link.exe /no
logo @CMakeFiles\rtaudio.dir\objects1.rsp /out:rtaudio.dll /implib:rtaudio.lib /pdb:C:\luaGL\gitsources\BUILDS
\anima_build_nmake32\LuaJIT-rtaudio\rtaudio\rtaudio.pdb /dll /version:6.0 /machine:X86 /debug /INCREMENTAL /DE
F:CMakeFiles\rtaudio.dir\exports.def dsound.lib ksuser.lib mfplat.lib mfuuid.lib wmcodecdspuuid.lib winmm.lib
ole32.lib kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.
lib advapi32.lib /MANIFEST /MANIFESTFILE:CMakeFiles\rtaudio.dir/intermediate.manifest CMakeFiles\rtaudio.dir/m
anifest.res" failed (exit code 1120) with the following output:
exports.def : error LNK2001: sÝmbolo externo = sin resolver
rtaudio.lib : fatal error LNK1120: 1 externos sin resolver
NMAKE : fatal error U1077: 'C:\cmake-3.13.0-rc1-win64-x64\bin\cmake.exe' : código devuelto '0xffffffff'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.15
.26726\bin\HostX86\x86\nmake.exe"' : código devuelto '0x2'
Stop.
NMAKE : fatal error U1077: '"C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Tools\MSVC\14.15
.26726\bin\HostX86\x86\nmake.exe"' : código devuelto '0x2'
Stop.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,7 +25,6 @@ endif()
 
 # Necessary for Windows
 if(WIN32)
-  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
   set(CMAKE_DEBUG_POSTFIX "d")
 endif()
 
@@ -197,6 +196,9 @@ if (NEED_PTHREAD)
   list(APPEND LINKLIBS Threads::Threads)
 endif()
 
+# Set compile-time definitions
+add_definitions(${API_DEFS})
+
 # Create library targets.
 cmake_policy(SET CMP0042 OLD)
 set(LIB_TARGETS)
@@ -216,7 +218,6 @@ if(BUILD_SHARED_LIBS)
     ${INCDIRS})
 
   # Set compile-time definitions
-  target_compile_definitions(rtaudio PRIVATE ${API_DEFS})
   target_compile_definitions(rtaudio PRIVATE RTAUDIO_EXPORT)
 
   target_link_libraries(rtaudio ${LINKLIBS})
@@ -236,9 +237,6 @@ if(BUILD_STATIC_LIBS)
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
     ${INCDIRS})
-
-  # Set compile-time definitions
-  target_compile_definitions(rtaudio_static PRIVATE ${API_DEFS})
 
   target_link_libraries(rtaudio_static ${LINKLIBS})
 endif()

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -28,6 +28,10 @@ if(WIN32)
   set(CMAKE_DEBUG_POSTFIX "d")
 endif()
 
+if(MINGW)
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
+
 # Build Options
 option(BUILD_SHARED_LIBS "Compile library shared lib." TRUE)
 option(BUILD_STATIC_LIBS "Compile library static lib." TRUE)
@@ -196,9 +200,6 @@ if (NEED_PTHREAD)
   list(APPEND LINKLIBS Threads::Threads)
 endif()
 
-# Set compile-time definitions
-add_definitions(${API_DEFS})
-
 # Create library targets.
 cmake_policy(SET CMP0042 OLD)
 set(LIB_TARGETS)
@@ -218,6 +219,7 @@ if(BUILD_SHARED_LIBS)
     ${INCDIRS})
 
   # Set compile-time definitions
+  target_compile_definitions(rtaudio PRIVATE ${API_DEFS})
   target_compile_definitions(rtaudio PRIVATE RTAUDIO_EXPORT)
 
   target_link_libraries(rtaudio ${LINKLIBS})
@@ -237,6 +239,9 @@ if(BUILD_STATIC_LIBS)
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
     $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
     ${INCDIRS})
+
+  # Set compile-time definitions
+  target_compile_definitions(rtaudio_static PRIVATE ${API_DEFS})
 
   target_link_libraries(rtaudio_static ${LINKLIBS})
 endif()

--- a/tests/playraw.cpp
+++ b/tests/playraw.cpp
@@ -44,7 +44,7 @@ typedef double  MY_TYPE;
 */
 
 // Platform-dependent sleep routines.
-#if defined( __WINDOWS_ASIO__ ) || defined( __WINDOWS_DS__ ) || defined( __WINDOWS_WASAPI__ )
+#if defined( WIN32 )
   #include <windows.h>
   #define SLEEP( milliseconds ) Sleep( (DWORD) milliseconds ) 
 #else // Unix variants

--- a/tests/playsaw.cpp
+++ b/tests/playsaw.cpp
@@ -41,7 +41,7 @@ typedef double MY_TYPE;
 */
 
 // Platform-dependent sleep routines.
-#if defined( __WINDOWS_ASIO__ ) || defined( __WINDOWS_DS__ ) || defined( __WINDOWS_WASAPI__ )
+#if defined( WIN32 )
   #include <windows.h>
   #define SLEEP( milliseconds ) Sleep( (DWORD) milliseconds ) 
 #else // Unix variants

--- a/tests/record.cpp
+++ b/tests/record.cpp
@@ -38,7 +38,7 @@ typedef double MY_TYPE;
 */
 
 // Platform-dependent sleep routines.
-#if defined( __WINDOWS_ASIO__ ) || defined( __WINDOWS_DS__ ) || defined( __WINDOWS_WASAPI__ )
+#if defined( WIN32 )
   #include <windows.h>
   #define SLEEP( milliseconds ) Sleep( (DWORD) milliseconds ) 
 #else // Unix variants

--- a/tests/teststops.cpp
+++ b/tests/teststops.cpp
@@ -21,7 +21,7 @@
 #define REPETITIONS 10
 
 // Platform-dependent sleep routines.
-#if defined( __WINDOWS_ASIO__ ) || defined( __WINDOWS_DS__ ) || defined( __WINDOWS_WASAPI__ )
+#if defined( WIN32 )
   #include <windows.h>
   #define SLEEP( milliseconds ) Sleep( (DWORD) milliseconds ) 
 #else // Unix variants

